### PR TITLE
Replace terrafile

### PR DESCRIPTION
**Pull request by neillturner**
## Context
Replace terrafile as it's not maintained anymore. Using simple git instead.

## Changes proposed in this pull request
- Update makefile to remove references to terrafile 
- Update makefile to add new vendor-modules target with git command to correct location for terraform code
- In global_config directory  add correct value for TERRAFORM_MODULES_TAG in config files 
- in terraform/application/config remove terrafile files 
- In terraform/domains directory remove terrafile files 
- Update .gitignore  remove bin/terrafile

## Guidance to review
i tested for each environment and to see terraform correctly get the module code
make development terraform-plan DOCKER_IMAGE_TAG=xxx
make production terraform-plan DOCKER_IMAGE_TAG=xxx

make domains-infra-plan
make development domains-plan
make production domains-plan

### Link to Trello card
[Replace terrafile](https://trello.com/c/5jCq2a3p/2158-replace-terrafile)

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by neillturner** [comment]
[Replace terrafile](https://trello.com/c/5jCq2a3p/2158-replace-terrafile)

File: Makefile [hunk 1]
```diff
@@ -68,9 +67,13 @@ deploy-arm-resources: arm-deployment
 
 validate-arm-resources: set-what-if arm-deployment
 
-domains-infra-init: domains set-azure-account
-	./bin/terrafile -p terraform/domains/infrastructure/vendor/modules -f terraform/domains/infrastructure/config/zones_Terrafile
+.PHONY: vendor-domain-infra-modules
+vendor-domain-infra-modules:
+	rm -rf terraform/domains/infrastructure/vendor/modules/domains
+	TERRAFORM_MODULES_TAG=stable
```
Comment (saliceti) [inline Makefile:73]: This variable is not set. You probably want to add this to domains.sh instead.

Comment (saliceti) [inline Makefile:73]: This line can be removed

Comment (neillturner) [reply Makefile]: removed

**Pull request merged**